### PR TITLE
Fix #289: Notebook 5.0 compatibility

### LINKED ISSUE #289: CLI quick-setup incompatibility with notebook 5.0
The BaseNBExtensionApp class is gone so the check if we're in >=4.2 fails here: https://github.com/jupyter/dashboards/blob/master/jupyter_dashboards/extensionapp.py#L12. Probably need to use notebook.extensions.BaseExtensionApp instead. That or remove the quick-setup convenience step and have people write out the install/enable/disable/uninstall manually.

https://github.com/jupyter/notebook/blob/master/notebook/extensions.py#L52

Ref: https://github.com/jupyter/notebook/issues/2150

### PR DESCRIPTION
**Pull request by parente**
* Switch to conda dev setup
* Fix quick-setup process for nb 5.0
* Fix selected cell CSS
* Fix bokeh demo for 0.12.4
* Cleanup documentation for 5.0

### CODE REVIEW

**Comment by parente** [comment]
@dtaeib, @jhpedemonte, @dalogsdon care to give this a look before I pull the trigger? It's all dev setup / install cleanup.

File: requirements-demo.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+bokeh==0.12.4
+ipywidgets
+matplotlib
+pandas
+seaborn
+widgetsnbextension
```
Comment (jhpedemonte) [inline requirements-demo.txt]: Should any other reqs (in this file and others) also be version limited?

File: Dockerfile.dev [hunk 2]
```diff
@@ -1,19 +0,0 @@
...
-    apt-get install -y curl && \
-    curl --silent --location https://deb.nodesource.com/setup_0.12 | sudo bash - && \
-    apt-get install --yes nodejs && \
-    npm install -g bower
-
-USER jovyan
-
-# Setup a sphinx environment for doc building
-COPY docs/environment.yml /tmp/environment.yml
-RUN conda env create -f /tmp/environment.yml
```
Comment (jhpedemonte) [inline Dockerfile.dev]: So getting rid of Docker in favor of Conda: just easier/cleaner? Or is there another reason?

**Comment by jhpedemonte** [comment]
Any idea what travis failure is about (for last commit)?

File: Dockerfile.dev (hunk 2 above)
Comment (parente) [inline Dockerfile.dev]: Consistency with other Python projects mainly.

File: requirements-demo.txt (hunk 1 above)
Comment (parente) [inline requirements-demo.txt]: If they break on newer versions, better to get issues about them.  As it stood before all these updates, the versions used were quite old from a pinned docker image. Great for a dev env, but it gives no indication of what the real install experience is going to be for users.

**Comment by parente** [comment]
I can't get the travis log to load at the moment. Seems have gone haywire since the S3 outage yesterday.

**Comment by jhpedemonte** [comment]
I get same error as in travis. When running `make env`, I get this:

```
source activate dashboards && \
		pip install -r requirements-doc.txt && \
		npm install && \
		npm run bower && \
		pip install -e . && \
		jupyter dashboards quick-setup --sys-prefix &&
/bin/bash: -c: line 1: syntax error: unexpected end of file
```

**Comment by jhpedemonte** [comment]
Working for the most part, except it can't find declarative widgets. Should that be added to one of the requirements files?

**Comment by parente** [comment]
Added decl widgets but it's currently busted on install with matplotlib 2.0. Opened an issue about it on the widget repo.

I think I addressed all the issues noted here. Going to merge this in prep for 5.0.

**Pull request merged**